core/trino-main/src/test/java/io/trino/operator/scalar/TestArrayHistogramFunction.java

```diff
@@ -0,0 +1,176 @@
+
+        assertThat(assertions.expression("array_histogram(a)")
+                .binding("a", "ARRAY['a', 'b', 'a']"))
+                .isEqualTo(Map.of("a", 2L, "b", 1L));
+    }
+
+    @Test
+    public void testDuplicateKeys()
+    {
+        assertThat(assertions.expression("array_histogram(a)")
+                             .binding("a", "ARRAY[42, 42, 42]"))
+                .isEqualTo(Map.of(42, 3L));
+
+        assertThat(assertions.expression("array_histogram(a)")
+                             .binding("a", "ARRAY['a', 'b', 'a']"))
+                .isEqualTo(Map.of("a", 2L, "b", 1L));
+    }
+
+    @Test
+    public void testEmpty()
+    {
+        assertThat(assertions.expression("array_histogram(a)")
+                .binding("a", "ARRAY[]"))
+                .isEqualTo(Map.of());
+    }
+
+    @Test
+    public void testNullsIgnored()
+    {
+        assertThat(assertions.expression("array_histogram(a)")
+                .binding("a", "ARRAY[NULL]"))
+                .isEqualTo(Map.of());
+
+        assertThat(assertions.expression("array_histogram(a)")
+                .binding("a", "ARRAY[42, NULL]"))
+                .isEqualTo(Map.of(42, 1L));
+
+        assertThat(assertions.expression("array_histogram(a)")
+                             .binding("a", "ARRAY[NULL, NULL, NULL, NULL, NULL]"))
+                .isEqualTo(Map.of());
+    }
+
+    @Test
+    public void testLargeArray()
+    {
+        assertThat(assertions.expression("array_histogram(a)")
+                             .binding("a", "ARRAY[42, 42, 42, 7, 1, 7, 1, 42, 7, 1, 1, 42, 7, 42, 1, 7, 2, 3, 7, 42, 42]"))
+                .isEqualTo(Map.of(1, 5L, 2, 1L, 3, 1L, 7, 6L, 42, 8L));
+    }
+
+    @Test
+    public void testEdgeCaseValues()
+    {
+        assertThat(assertions.expression("array_histogram(a)")
+                             .binding("a", "ARRAY[NULL, '', NULL, '']"))
+                .matches("MAP(ARRAY[''], CAST(ARRAY[2] AS ARRAY(BIGINT)))");
+
+        assertThat(assertions.expression("array_histogram(a)")
+                             .binding("a", "CAST(ARRAY[NULL, -0.1, 0, 0.1, -0.1, NULL, 0.0] AS ARRAY(DECIMAL(1,1)))"))
+                .matches("MAP(CAST(ARRAY[-0.1, 0.0, 0.1] AS ARRAY(DECIMAL(1,1))), CAST(ARRAY[2, 2, 1] AS ARRAY(BIGINT)))");
+
+        assertThat(assertions.expression("array_histogram(a)")
+                             .binding("a", "ARRAY[IPADDRESS '10.0.0.1', IPADDRESS '::ffff:a00:1']"))
+                .matches("MAP(ARRAY[IPADDRESS '::ffff:a00:1'], CAST(ARRAY[2] AS ARRAY(BIGINT)))");
+
+        assertThat(assertions.expression("transform_keys(array_histogram(a), (k, v) -> k AT TIME ZONE 'UTC')")
+                             .binding("a", "ARRAY[TIMESTAMP '2001-01-01 01:00:00.000 UTC', TIMESTAMP '2001-01-01 02:00:00.000 +01:00']"))
+                .matches("MAP(ARRAY[TIMESTAMP '2001-01-01 01:00:00.000 UTC'], CAST(ARRAY[2] AS ARRAY(BIGINT)))");
+    }
+
+    @Test
+    public void testTypes()
```

Comment:
You could use `org.testng.annotations.DataProvider` to avoid repeating the same query over and over.

Reply:
No, don’t use that. It makes it a lot harder to set breakpoints and debug when there are issues.